net-irc/irssi/patches/irssi-0.8.17-haiku.patch

```diff
@@ -0,0 +1,350 @@
+         WINDOW_REC *old_active_win;
+ 	GSList *tmp;
+ 	char *str;
++	SBAR_ITEM_REC *rec;
+
+ 	old_active_win = active_win;
+         if (bar->parent_window != NULL)
+@@ -977,7 +979,7 @@ static void statusbar_redraw_needed_items(STATUSBAR_REC *bar)
+ 	}
+
+ 	for (tmp = bar->items; tmp != NULL; tmp = tmp->next) {
+-		SBAR_ITEM_REC *rec = tmp->data;
++		rec = (SBAR_ITEM_REC *) tmp->data;
+
+ 		if (rec->dirty ||
+ 		    (bar->dirty_xpos != -1 &&
+@@ -994,6 +996,7 @@ static void statusbar_redraw_needed_items(STATUSBAR_REC *bar)
+ void statusbar_redraw_dirty(void)
+ {
+ 	GSList *tmp;
++	STATUSBAR_REC *rec;
+
+ 	if (statusbar_need_recreate_items) {
+ 		statusbar_need_recreate_items = FALSE;
+@@ -1001,7 +1004,7 @@ void statusbar_redraw_dirty(void)
+ 	}
+
+ 	for (tmp = active_statusbar_group->bars; tmp != NULL; tmp = tmp->next) {
+-		STATUSBAR_REC *rec = tmp->data;
++		rec = (STATUSBAR_REC *) tmp->data;
+
+ 		if (rec->dirty) {
+                         statusbar_redraw_needed_items(rec);
+@@ -1021,9 +1024,10 @@ void statusbar_redraw_dirty(void)
+ static void statusbars_remove_unvisible(MAIN_WINDOW_REC *window)
+ {
+ 	GSList *tmp, *next;
++	STATUSBAR_REC *bar;
+
+ 	for (tmp = window->statusbars; tmp != NULL; tmp = next) {
+-		STATUSBAR_REC *bar = tmp->data;
++		bar = (STATUSBAR_REC *) tmp->data;
+
+ 		next = tmp->next;
+                 if (!STATUSBAR_IS_VISIBLE(bar->config, window))
+@@ -1036,10 +1040,11 @@ static void statusbars_add_visible(MAIN_WINDOW_REC *window)
+ 	STATUSBAR_GROUP_REC *group;
+         STATUSBAR_REC *bar;
+ 	GSList *tmp;
++	STATUSBAR_CONFIG_REC *config;
+
+         group = active_statusbar_group;
+ 	for (tmp = group->config_bars; tmp != NULL; tmp = tmp->next) {
+-		STATUSBAR_CONFIG_REC *config = tmp->data;
++		config = (STATUSBAR_CONFIG_REC *) tmp->data;
+
+ 		if (config->type == STATUSBAR_TYPE_WINDOW &&
+ 		    STATUSBAR_IS_VISIBLE(config, window) &&
+--
+2.2.2
+
+From 8a4b09aaf54e25d693f3c52d27181e38797773f9 Mon Sep 17 00:00:00 2001
+From: pisculichi <pisculichi@openmailbox.org>
+Date: Fri, 18 Dec 2015 00:10:45 +0000
+Subject: [PATCH] Support haiku socket
+
+---
+ configure.ac | 14 ++------------
+ 1 file changed, 2 insertions(+), 12 deletions(-)
+
+diff --git a/configure.ac b/configure.ac
+index f148885..c534a3e 100644
+--- a/configure.ac
++++ b/configure.ac
+@@ -196,18 +196,8 @@ case "$host_os" in
+     ;;
+ esac
+
+-
+-AC_CHECK_FUNC(socket, [], [
+-	AC_CHECK_LIB(socket, socket, [
+-		LIBS="$LIBS -lsocket"
+-	])
+-])
+-
+-AC_CHECK_FUNC(inet_addr, [], [
+-	AC_CHECK_LIB(nsl, inet_addr, [
+-		LIBS="$LIBS -lnsl"
+-	])
+-])
++AC_SEARCH_LIBS([socket], [socket])
++AC_SEARCH_LIBS([inet_addr], [nsl])
+
+ dnl * gcc specific options
+ if test "x$ac_cv_prog_gcc" = "xyes"; then
+--
+2.2.2
+
+From cfff5e78aba54d0f881f212080c2beb7bc090bea Mon Sep 17 00:00:00 2001
+From: pisculichi <pisculichi@openmailbox.com>
+Date: Sun, 27 Mar 2016 15:04:26 +0000
+Subject: [PATCH] Add links to convert html -> text in autogen.sh
```

Comment:
if you do not want to autogen yourself you can skip converting the docs altogether, since this step has already been done in release tarballs.